openpype/plugins/publish/extract_burnin.py

```diff
@@ -687,10 +712,13 @@ def prepare_repre_data(self, instance, repre, burnin_data, temp_data):
 
         burnin_duration = burnin_frame_end - burnin_frame_start + 1
 
+        frame_id = burnin_duration if not keep_frame_index else burnin_frame_end
```

Comment:
Why the `else` value is the end frame value ?

Reply:
Because when publishing with a custom_frame or keeping the index, then frame_id used in otio_brunin must go all the way to the end of the given index and not the automated recalculate frame index
Basically, OP rname the output files starting from 1 to the index on the markOut, even if the marks in tvpp are like [25 - 60]
Without it, OP will concidere the frame_id to be 36 (60-24) to match [1-36] and not [25-60]

Plus if the user enter the custom_frames [25, 30, 60], the duration is 3, and OP would only burnIn the first 3 images and won't go to the end.

You can search the uasge of frame_id in otio_burin.py

Reply:
I'm on it to simplify it